kernel/core/datamanagement/dataaccess.cpp

```diff
@@ -1852,6 +1853,91 @@ static std::string tableMethod_anova(const std::string& sTableName, std::string
     return _kernel->getStringParser().createTempStringVectorVar(vRet);;
 }
 
+/////////////////////////////////////////////////
+/// \brief Realizes the "kmeansof()" table method.
+///
+/// \param sTableName const std::string&
+/// \param sMethodArguments std::string
+/// \param sResultVectorName const std::string&
+/// \return std::string
+///
+/////////////////////////////////////////////////
+static std::string tableMethod_kmeans(const std::string& sTableName, std::string sMethodArguments, const std::string& sResultVectorName)
+{
+    NumeReKernel* _kernel = NumeReKernel::getInstance();
+
+    std::string sCols = getNextArgument(sMethodArguments, true);
+    //sCols += "," + getNextArgument(sMethodArguments, true);
+
+    int nResults = 0;
+    _kernel->getMemoryManager().updateDimensionVariables(sTableName);
+    _kernel->getParser().SetExpr(sCols);
+    mu::value_type* v = _kernel->getParser().Eval(nResults);
+
+    if (nResults < 1)
+        throw SyntaxError(SyntaxError::TOO_FEW_COLS, sTableName + "().kmeansof()", ".kmeansof(", ".kmeansof(");
+
+    auto cols = VectorIndex(v, nResults, 0);
+
+    sCols = getNextArgument(sMethodArguments, true);
+    nResults = 0;
+    _kernel->getMemoryManager().updateDimensionVariables(sTableName);
+    _kernel->getParser().SetExpr(sCols);
+    v = _kernel->getParser().Eval(nResults);
+
+    if (nResults < 1)
+        throw SyntaxError(SyntaxError::TOO_FEW_COLS, sTableName + "().kmeansof()", ".kmeansof(", ".kmeansof(");
+
+    size_t nClusters = intCast(v[0]);
+
+    VectorIndex vIndex(0, VectorIndex::OPEN_END);
+
+    size_t maxIterations = 100;
+    size_t init_method = 1;
+    size_t n_init = 10;
+
+    if (sMethodArguments.length())

```

Comment:
If all value are numerical, you can also calculate all of them at the same time (just pipe in the whole expression) and evaluate the number of return values. That might be more elegant and easier to read.